From: hololeap <hololeap@protonmail.com>
Date: Thu, 14 Mar 2024 15:37:37 -0600
Subject: [PATCH 1/1] Fix tests

Tests try to incorrectly use fromList, instead of the correct
arrayFromList imported from Data.Primitive.Array in the primitive
package.

Signed-off-by: hololeap <hololeap@protonmail.com>
--- a/test/Main.hs
+++ b/test/Main.hs
@@ -55,10 +55,10 @@ lawsToTest :: QCC.Laws -> TestTree
 lawsToTest (QCC.Laws name pairs) = testGroup name (map (uncurry TQC.testProperty) pairs)
 
 instance Arbitrary1 Array where
-  liftArbitrary elemGen = fmap fromList (QC.liftArbitrary elemGen)
+  liftArbitrary elemGen = fmap arrayFromList (QC.liftArbitrary elemGen)
 
 instance Arbitrary a => Arbitrary (Array a) where
-  arbitrary = fmap fromList QC.arbitrary
+  arbitrary = fmap arrayFromList QC.arbitrary
 
 instance Arbitrary1 SmallArray where
   liftArbitrary elemGen = fmap smallArrayFromList (QC.liftArbitrary elemGen)
-- 
2.43.2

